### PR TITLE
Adds alien jobbans and adds jobban and pref support to various events

### LINKED ISSUE #2486: Can't ban people from Alien
##### Revision that this bug was encountered on (REQUIRED!! Type 'show-server-revision' ingame to find out - note that 'current revision' isn't helpful!): #####
Last pull was 2461.

##### Which server were you playing on? (REQUIRED!! e.g. TG-Sibyl1, TG-Sibyl2, Badgerstation, Facepunch, Pinkstation, /VG/... etc.) #####
Sybil, unmodded tg code.

##### Problem Description #####
You can't ban people from being aliens.
###### What did you expect to happen? ######
To be able to ban aliens who kill their own.

###### What happened instead? ######
You can't!

###### Why is this bad/What are the consequences? ######
Can't ban alien shitlers.

##### Steps to reproduce the problem (remember to be specific! Those little details can save us hours of work!): #####
1. Try to ban from alien.
2. Fail.

##### Possibly related stuff (which gamemode was it? What were you doing at the time? Was anything else out of the ordinary happening?): #####
Nope.

##### Anything else? #####
Sentinel 10 please get fucked.

### LINKED ISSUE #12472: nukeops ban doesn't ban you for syndie borg
ayy.lmao

### PR DESCRIPTION
**Pull request by Shadowlight213**
Fixes #12472 
Fixes #2486
Adds feedback to alien infestation event
Changes centcom official to use deathsquad jobban
Syndieborgs now use operative jobban
Support for antag candidate prefs added to various events to reduce popup spam.
Adds Shadowling Jobbans

### CODE REVIEW

File: code/__HELPERS/game.dm [hunk 1]
```diff
@@ -298,7 +298,9 @@
 			if(G.client != null)
 				if(!(G.mind && G.mind.current && G.mind.current.stat != DEAD))
 					if(!G.client.is_afk(afk_bracket) && (G.client.prefs.be_special & be_special_flag))
-						candidates += G.client
+						if (jobbanType)
```
Comment (RemieRichards) [inline code/__HELPERS/game.dm]: it should still add the client if there is no ```jobbanType```

Comment (Shadowlight213) [inline code/__HELPERS/game.dm]: Fixed

File: code/__HELPERS/game.dm [hunk 2]
```diff
@@ -298,7 +298,11 @@
 			if(G.client != null)
 				if(!(G.mind && G.mind.current && G.mind.current.stat != DEAD))
 					if(!G.client.is_afk(afk_bracket) && (G.client.prefs.be_special & be_special_flag))
-						candidates += G.client
+						if (jobbanType)
+							if(!(jobban_isbanned(G, jobbanType) || jobban_isbanned(G, "Syndicate")))
+								candidates += G.client
```
Comment (optimumtact) [inline code/__HELPERS/game.dm]: why the specific check for syndicates?

File: code/modules/admin/topic.dm [hunk 3]
```diff
@@ -802,13 +808,13 @@
 			jobs += "<td width='20%'><a href='?src=\ref[src];jobban3=malf AI;jobban4=\ref[M]'><font color=red>[replacetext("Malf AI", " ", "&nbsp")]</font></a></td>"
 		else
 			jobs += "<td width='20%'><a href='?src=\ref[src];jobban3=malf AI;jobban4=\ref[M]'>[replacetext("Malf AI", " ", "&nbsp")]</a></td>"
-
+*/
```
Comment (optimumtact) [inline code/modules/admin/topic.dm]: commented out code detected

File: code/__HELPERS/game.dm (hunk 2 above)
Comment (Jordie0608) [inline code/__HELPERS/game.dm]: IIRC jobbanning someone from Syndicate is how the system handles banning someone from all antag roles at once.

File: code/modules/admin/topic.dm (hunk 3 above)
Comment (Shadowlight213) [inline code/modules/admin/topic.dm]: That whole block was commented out originally. I just uncommented the alien part

**Pull request merged**